chatter/src/components/UserList/UserList.js

```diff
@@ -1,12 +1,23 @@
-import React, { useContext } from 'react';
+import React, { useContext, memo, useState } from 'react';
 import cx from 'classnames';
 import LatestMessagesContext from '../../contexts/LatestMessages/LatestMessages';
 import UserProfile from '../../common/components/UserProfile/UserProfile';
 import USERS from './constants/users';
 import './_user-list.scss';
 
-function User({ icon, name, lastActive, isOnline, userId, color }) {
-  const { messages } = useContext(LatestMessagesContext);
+const User = memo(({ icon, name, lastActive, isOnline, userId, color }) => {
+  const { lastMessageSource$ } = useContext(LatestMessagesContext);
+  const [message, setMessage] = useState();
+  
+  // Only subscribe from the global observable to make sure only the impacted user is re-rendered.
+  React.useEffect(() => {
+    // Subscribe global latest messages change.
+    const sub$ = lastMessageSource$.subscribe(messages => {
+      setMessage(messages[userId])
+    })
+
+    return () => sub$.unsubscribe();
+  }, []);
```

Comment:
Above change can make sure only the user `Bot`'s latest message gets re-rendered. Before, all users got re-rendered whenever new bot message came in.